### PR TITLE
Manual  PR 2 The Electric Boogaloo

### PR DESCRIPTION
**Pull request by david-samuel-mcdermott**

### CODE REVIEW

**Comment by ThePBJain** [comment]
Just a general question? What's the dia?

File: manual/manual.tex [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+\title{HackPSU RFID Scanner Box User Manual}
+\date{\today}
+
+\begin{document}
+\maketitle
+
+\section*{Introduction}
+This document serves as the reference manual for the HackPSU RFID scanner box.  This manual does not detail the implementation of the device; the intention of this document is to provide a user with the necessary information to operate the scanner and to troubleshoot some basic problems.
+
```
Comment (ThePBJain) [inline manual/manual.tex]: Could we have this information as a README.md in the root of the repo as well?

File: manual/manual.tex [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+\usepackage{graphicx}
+
+\title{HackPSU RFID Scanner Box User Manual}
+\date{\today}
+
+\begin{document}
+\maketitle
+
+\section*{Introduction}
+This document serves as the reference manual for the HackPSU RFID scanner box.  This manual does not detail the implementation of the device; the intention of this document is to provide a user with the necessary information to operate the scanner and to troubleshoot some basic problems.
```
Comment (ThePBJain) [inline manual/manual.tex]: We should eventually have a doc detailing the implementation of the device too! Also cost of building each as well. Probably something we can write up after the hackathon.

File: manual/manual.tex [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	\item[\textbf{BAUD\_RATE}] The baud rate for the serial communications.  
+	\item[\textbf{WIFI\_CONSTS}] This definition should be empty; it is a flag marking whether or not the SSID and PASSWD fields have been defined; if unset, the sketch will fail to compile. 
+	\item[\textbf{REDIS}] The address of the cache server; this should not have any protocol, just a network address and a port.
+	\item[\textbf{SSID}] The name of the network to connect to; if this network cannot be found, the scanner will not boot.
+	\item[\textbf{PASSWD}] The password for the network that is being connected to; if this is incorrect, the scanner will not boot.
+	\item[\textbf{fp}] The fingerprint of the SSL certificate being used by the caching server.
+\end{itemize}
+
+\section{Troubleshooting}
+
```
Comment (ThePBJain) [inline manual/manual.tex]: We should add a subsection on garbage outputs to screens.

File: manual/manual.tex [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+	\item[\textbf{C}] The 'C' key serves no special functionality presently.  In a future implementation, it will scroll the LCD.
+	\item[\textbf{D}] The 'D' key transitions the box into the locked state.
+	\item[\textbf{*}] The '*' key acts as a clear or negative input button.  
+	\item[\textbf{\#}] The '\#' key acts as a submit or positive input button.
+\end{itemize}
+
+\section{Access Control}
+Access control is handled in a locked state, where the scanner waits for a staff wristband to be scanned.  This is intended to provide a minimal layer of security on the device so that event goers cannot scan themselves into events.  The duplicate state will produce a staff band by writing the master key into the data sector of the band.  Access control can be disabled in the implementation.
+
+\section{Scanning}
```
Comment (sushshring) [inline manual/manual.tex]: Outline this as the most important state of the scanner.

File: manual/manual.tex [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+Upon entering the "Locate \& Scan" state, the scanner will fetch a list of available locations from the cache server defined in the configuration header.  To select an entry from the menu, press the '\#' key. Numerical inputs are not supported in this menu.  Once the location is selected; wristbands may be scanned.  The scanner is set with a timeout, which cycles the functionality to listen for keypad driven state transitions using the 'B' and 'D' keys.  Upon receiving a successful scan, a request is sent to the cache server to commit the scan.  If the response is received and the "isRepeat" field is set to 'false' then the LCD will display "Allow" if no response is received or if the field is set to 'true' then "Deny" will be printed to the LCD.  The only case where the "isRepeat" field will be set to 'true' is if the event at the current location is a 'food' event and the user has been scanned once before.
```
Comment (sushshring) [inline manual/manual.tex]: `The scanner is set with a timeout, which cycles the functionality to listen for keypad driven state transitions using the 'B' and 'D' keys.` Can you simplify this?

File: manual/manual.tex [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
...sing the keypad.  Once a pin is submitted, a request to retrieve the user data will be made.  If no response is received or if the pin is not found on the cache server, the message "Invalid Pin" will be displayed.  Upon successfully entering a pin, a new prompt will appear, asking the user to validate the name displayed below.  The user may press '\#' to confirm that the user pin was correct, or '*' to return to pin entry.  The user will then be prompted to scan a wristband.  If the wristband's UID already exists in the cache server's database, then a message instructing the user to dispose of the band will be displayed.  If the scan function times out before a wristband is scanned, the pin will need to be entered again. If the user name is confirmed and a wristband is scanned, the user will be checked-in on the cache server.  The next step displays the user's shirt size and asks for photo consent.  Any key can be pressed to escape the photo consent menu, although using '*' or '\#' is recommended as this may change in a future implementation.  The scanner will then return to the beginning of the check-in process.  Press 'B' or 'D' in the pin entry state to leave the check-in state.
```
Comment (sushshring) [inline manual/manual.tex]: Maybe `\begin{itemize}` the steps?

File: manual/manual.tex [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+	\item[\textbf{WIFI\_CONSTS}] This definition should be empty; it is a flag marking whether or not the SSID and PASSWD fields have been defined; if unset, the sketch will fail to compile. 
+	\item[\textbf{REDIS}] The address of the cache server; this should not have any protocol, just a network address and a port.
+	\item[\textbf{SSID}] The name of the network to connect to; if this network cannot be found, the scanner will not boot.
+	\item[\textbf{PASSWD}] The password for the network that is being connected to; if this is incorrect, the scanner will not boot.
+	\item[\textbf{fp}] The fingerprint of the SSL certificate being used by the caching server.
+\end{itemize}
+
+\section{Troubleshooting}
+
+\subsection{Long Scan Times}
```
Comment (sushshring) [inline manual/manual.tex]: Unresponsive scans also could mean that the RFID was already scanned. Maybe there should be a garbage band with every box to figure that out?

File: manual/manual.tex [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+If scan times are taking a long time, the likely culprit is a bad RFID signal or a long RTT on the connection to the cache server.  A bad RFID signal may be caused by excess material between the wristband and the scanner of the box.  A long RTT can be caused by long network hops to the cache server or by network congestion.  To test RTT with your cache server, use the following "time\_total" feature in cURL. 
+
+\subsection{Failure to Boot}
```
Comment (sushshring) [inline manual/manual.tex]: Again an itemized list for point of failures here would make it easier to read

File: manual/manual.tex (hunk 6 above)
Comment (mcderdav) [reply manual/manual.tex]: Good call, itemizing now

File: manual/manual.tex (hunk 7 above)
Comment (mcderdav) [reply manual/manual.tex]: Good catch.  For this event, we are assuming that all operators have a staff band, so I will add that catch

File: manual/manual.tex [hunk 9]
```diff
@@ -1,20 +1,22 @@
...
 
 \usepackage{graphicx}
+\usepackage{hyperref}
 
 \title{HackPSU RFID Scanner Box User Manual}
 \date{\today}
+\author{ \href{mailto:technology@hackpsu.org}{HackPSU Technology Team}}
 
 \begin{document}
 \maketitle
```
Comment (sushshring) [inline manual/manual.tex]: Can you also add \tableofcontents

File: manual/manual.tex [hunk 10]
```diff
@@ -32,9 +34,41 @@ \section{Access Control}
...
+\subsection{User Guide}
+\begin{itemize}
+	\item The box will request that a registrant's pin is entered
+	\item Use the keypad to enter the registrant's base 10 bin, or press 'D' to escape to the locked state
+	\item If the registrant's pin was invalid, a message will be displayed
+	\item If the registrant's pin was valid; it will ask that you confirm their name.  Use the '\#' key to confirm or the '*' key to deny
+	\item Once the registrant's name is confirmed, the box will be ready to assign them a UID from a wristband
+	\item Scan the wristband now, and if the box says to scrap the wristband, please discard the band as a UID collision has occurred
+	\item Once the wristband has been successfully scanned, the registrant's shirt size will be displayed and a prompt asking for photo consent will appear.
```
Comment (sushshring) [inline manual/manual.tex]: Add: Hand the registrant the appropriate shirt size. We don't really entertain shirt size change requests

File: manual/manual.tex [hunk 11]
```diff
@@ -51,8 +85,17 @@ \section{Configuration}
 
 \section{Troubleshooting}
 
+\subsection{Failure to Scan an RFID Tag}
+There are two reasons that the box may not scan an RFID Tag.  The most likely reason is that the given band was already scanned.  The scanner is stately and will prevent the same band from being scanned twice in a row.  This is done to prevent multiple unintentional scans, but may result in the scanner being soft locked.  To unlock the scanner, simply scan another wristband or enter the locked state.  It is suggested that each operator have a band with the MASTER\_KEY written to it to unlock the box from the locked state and to free the scanner from a soft lock.
```
Comment (sushshring) [inline manual/manual.tex]: nit: stateful?

**Comment by sushshring** [comment]
Added a few more comments, but with the quick start section this should be pretty good. We can include a print of the Quick Start with the organizer clipboards

File: manual/manual.tex (hunk 4 above)
Comment (mcderdav) [reply manual/manual.tex]: Added one line to mark `!important`

File: manual/manual.tex (hunk 9 above)
Comment (mcderdav) [reply manual/manual.tex]: Look literally two lines down below `\maketitle`

File: manual/manual.tex (hunk 10 above)
Comment (mcderdav) [reply manual/manual.tex]: Added human interaction bullet

File: manual/manual.tex (hunk 11 above)
Comment (mcderdav) [reply manual/manual.tex]: I am confusion, what would you like done?

Comment (sushshring) [reply manual/manual.tex]: oh ignore this one

**Pull request merged**